pkg/component/dependencywatchdog/access.go

```diff
@@ -67,42 +98,150 @@ type dependencyWatchdogAccess struct {
 
 // AccessValues contains configurations for the component.
 type AccessValues struct {
-	// ServerOutOfCluster is the out-of-cluster address of a kube-apiserver.
-	ServerOutOfCluster string
 	// ServerInCluster is the in-cluster address of a kube-apiserver.
 	ServerInCluster string
 }
 
 func (d *dependencyWatchdogAccess) Deploy(ctx context.Context) error {
+	// Create shoot access secret
 	caSecret, found := d.secretsManager.Get(v1beta1constants.SecretNameCACluster)
 	if !found {
 		return fmt.Errorf("secret %q not found", v1beta1constants.SecretNameCACluster)
 	}
 
-	for name, server := range map[string]string{
-		InternalProbeSecretName: d.values.ServerInCluster,
-		ExternalProbeSecretName: d.values.ServerOutOfCluster,
-	} {
-		var (
-			shootAccessSecret = gardenerutils.NewShootAccessSecret(name, d.namespace).WithNameOverride(name)
-			kubeconfig        = kubernetesutils.NewKubeconfig(
-				d.namespace,
-				clientcmdv1.Cluster{Server: server, CertificateAuthorityData: caSecret.Data[secretsutils.DataKeyCertificateBundle]},
-				clientcmdv1.AuthInfo{Token: ""},
-			)
+	var (
+		shootAccessSecret = gardenerutils.NewShootAccessSecret(KubeConfigSecretName, d.namespace).WithNameOverride(KubeConfigSecretName)
+		kubeconfig        = kubernetesutils.NewKubeconfig(
+			d.namespace,
+			clientcmdv1.Cluster{Server: d.values.ServerInCluster, CertificateAuthorityData: caSecret.Data[secretsutils.DataKeyCertificateBundle]},
+			clientcmdv1.AuthInfo{Token: ""},
 		)
+	)
 
-		if err := shootAccessSecret.WithKubeconfig(kubeconfig).Reconcile(ctx, d.client); err != nil {
-			return err
-		}
+	if err := shootAccessSecret.WithKubeconfig(kubeconfig).Reconcile(ctx, d.client); err != nil {
+		return err
+	}
+
+	// Delete old Secrets and CM with shoot reconcile
+	if err := kubernetesutils.DeleteObjects(ctx, d.client,
+		&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: InternalProbeSecretName, Namespace: d.namespace}},
+		&corev1.Secret{ObjectMeta: metav1.ObjectMeta{Name: ExternalProbeSecretName, Namespace: d.namespace}},
+		&corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: TempGRMConfigMapName, Namespace: d.namespace}},
+	); err != nil {
+		return err
+	}
```

Comment:
Let's move this into the `app.go` file as well so that we can get rid of this "migration code" after a few releases.

Reply:
I'd agree with moving deletion of `InternalProbeSecret` and `ExternalProbeSecret` in `app.go`, but I don't think we can do the same for `TempGRMConfigMap`
`TempGRMConfigMap` is the configmap created within `app.go` and is needed temporarily until the shoot is reconciled and the proper cm is then deployed 

What do you suggest here?

Reply:
The new `ConfigMap` should be made immutable (like the old/others). Then it will be automatically garbage-collected once no longer in use.